sympy/series/sequences.py

```diff
@@ -711,6 +712,137 @@ def coeff_mul(self, coeff):
         return SeqFormula(formula, self.args[1])
 
 
+class RecursiveSeq(SeqBase):
+
+    """A finite degree recursive sequence.
+
+    That is, a sequence a(n) that depends on a fixed, finite number of its
+    previous values. The general form is
+
+        a(n) = f(a(n - 1), a(n - 2), ..., a(n - d))
+
+    for some fixed, positive integer d, where f is some function defined by a
+    SymPy expression.
+
+    Examples
+    ========
+
+    >>> from sympy import Function, symbols
+    >>> from sympy.series.sequences import RecursiveSeq
+
+    >>> y = Function("y")
+    >>> n = symbols("n")
+    >>> fib = RecursiveSeq((0, 1), y(n), y(n - 1) + y(n - 2))
+
+    For value at a particular point
+
+    >>> fib.coeff(3)
+    2
+
+    supports slicing
+
+    >>> fib[:6]
+    [0, 1, 1, 2, 3, 5]
+
+    inspect recurrence
+
+    >>> fib.recurrence
+    y(n - 2) + y(n - 1)
```

Comment:
Would it be better to have `Eq(y(n), y(n - 2) + y(n - 1))`?

Reply:
Do you mean the initial recurrence argument or just returning the recurrence for inspection from `seq.recurrence`?

Reply:
I think this is a good idea. I changed the output to match this.